src/App.tsx

```diff
@@ -1,27 +1,61 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
+import { getGoods } from './api/goods';
+import { Good } from './types/Good';
+import { Color } from './types/Color';
 // or
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = React.memo(() => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [amount, setAmount] = useState(0);
+  const [color, setColor] = useState('');
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  useEffect(() => {
+    getGoods(amount, color)
+      .then(setGoods);
+  }, [amount, color]);
```

Comment:
remove `amount` and `color` state variables. try to create 3 separate function by using `useCallback`

Reply:
I still don`t understand how to use it..